components/ledger/pkg/ledger/execute_script.go

```diff
@@ -21,6 +21,13 @@ import (
 )
 
 func (l *Ledger) ExecuteScript(ctx context.Context, preview bool, script core.ScriptData) (core.ExpandedTransaction, error) {
+
+	unlock, err := l.locker.Lock(ctx, l.store.Name())
```

Comment:
Should we check that the locker is nil ? (Even if it's in the default options)

Reply:
I don't think so, il a kind of "defensive programming" :)
But i can if you prefer.

Reply:
No i'm good with keeping that like that if we are sure that it will never be nil !